### DISCUSSION TITLE
Scanning a code in Dependency track dash board

### DISCUSSION THREAD

**Discussion by VASAVI512** [DependencyTrack/dependency-track#1065] [General]
Hi Steve,

I am not finding the option to scan a product form dependency track dash board. could you show me how to scan from dash board. 

Also we are looking for a tool who does the scan of commercial codes and shows their vulnerabilities. Please provide your comments on this requirements whether this tool does the same.

Regards,
Vasavi

**Comment by stevespringett** [comment]
Every time an SBOM is imported into Dependency-Track, the inventory in the SBOM is analyzed on demand. Complete analysis of the entire portfolio occurs every 24 hours, regardless of whether an SBOM was imported into a project or not. There isn't a manual way to trigger this. Most DT use cases focus on automation.

There are plenty of tools (open source and commercial) that scan code and show vulnerabilities. Some of these tools also produce SBOMs in CycloneDX format. Scanning things is outside the scope of DT. That's not what DT is designed for. Refer to the animation on the website (desktop or large screen recommended) https://dependencytrack.org/

**Reply by VASAVI512** [reply]
Thank you for the confirmation Steve.  So we need to upload the SBOM(components) to DT, then DT does the vulnerability validation on those components on every 24 hours.

Also, is DT is able to capture the vulnerabilities of commercial components as well? Please let us know.

Regards,
Vasavi

**Reply by stevespringett** [reply]
>Also, is DT is able to capture the vulnerabilities of commercial components as well?

Yes. But this requires the components have a CPE defined in the SBOM. CPE is a deprecated spec and has a known chicken/egg issue. Its not possible to monitor a product for known vulnerabilities until that product has its first CVE. This is generally the first time a CPE will be defined for a product.  So if a commercial product has a CPE already, yes, you can monitor commercial software for known vulnerabilities.

**Reply by VASAVI512** [reply]
Thanks for the information. I have one  more query, CycloneDX is able to generate a SBOM for Binary files like, .dll, exe, .jar?

Also, how do we create a SBOM from the product source code which is  a combination of .cs (Dotnet code), .js (NPM ), C,C++, etc...

As i understood, CycloneDX creates a SBOM for the .net code bases on the .sln, csproj,, .config, these input file. Is it not possible to scan a code with out these files in the product code?

**Reply by stevespringett** [reply]
> Thanks for the information. I have one more query, CycloneDX is able to generate a SBOM for Binary files like, .dll, exe, .jar?

Of course. CycloneDX is just a spec. Its up to the implementations of the spec to be able to meet various use cases - e.g. binary scanning.

> Also, how do we create a SBOM from the product source code which is a combination of .cs (Dotnet code), .js (NPM ), C,C++, etc..

You'll likely need to create multiple SBOMs from the various ecosystems and merge them together. The CycloneDX CLI does a very good job at that.

>As i understood, CycloneDX creates a SBOM for the .net code bases on the .sln, csproj,, .config, these input file. Is it not possible to scan a code with out these files in the product code?

You may want to start asking CycloneDX specific questions in the CycloneDX Slack channel. I have only limited knowledge about each CycloneDX implementation. See https://cyclonedx.org/slack/invite

**Reply by VASAVI512** [reply]
Thank you so much for your complete information. i will also check with CycloneDX specific questions with CycloneDX Chaneel.

**Reply by VASAVI512** [reply]
Hello Steve,

Is it possible to generate the project report or any detailed from DT, if yes, how can we generate. 

Regards,
Vasavi

**Reply by stevespringett** [reply]
DT does not provide reporting capabilities. All data is available via APIs. Users are encouraged to adopt vulnerability aggregation platforms such as ThreadFix, CodeDx, or Kenna Security which are all supported by DT or to leverage general purpose BI reporting tools.

**Reply by VASAVI512** [reply]
Ok. Thank you.


I have downloaded the code from : https://github.com/aspnet/AspNetWebStack
I have created the SBOM using CycloneDX. 
I do not see the below type of license was not updated in Dependency Tracker Tool, why?.
Will it allow only SPDX license ids tags(Ex: <license>MIT<license>)?

 <component type="library">
      <publisher>Microsoft</publisher>
      <name>DotNetOpenAuth.AspNet</name>
      <version>4.0.3.12153</version>
      <description>Allow your web visitors to log into your web site using accounts they already have with popular services.</description>
      <hashes>
        <hash alg="SHA-512">8307767EF15610128D6F778DF0C349E025D6F68E818895C7042DFD998EC64483960DDAD48680D9DA6FED2CC44078FC630CA51DD6BD7916A4CAAC3443406DEF6A</hash>
      </hashes>
      <licenses>
        <license>
          <url>http://www.opensource.org/licenses/ms-pl.html</url>
        </license>
      </licenses>
      <purl>pkg:nuget/DotNetOpenAuth.AspNet@4.0.3.12153</purl>
      <externalReferences>
        <reference type="website">
          <url>http://www.dotnetopenauth.net/</url>
        </reference>
      </externalReferences>
    </component>

**Reply by VASAVI512** [reply]
`<component type="library">
      <publisher>Andrew Arnott</publisher>
      <name>DotNetOpenAuth.Core</name>
      <version>4.0.3.12153</version>
      <description>Shared library implementing the messaging pipeline used by auxiliary DotNetOpenAuth libraries.</description>
      <hashes>
        <hash alg="SHA-512">E2F45DD8B6321FF5E5EDFD9DBE1AD9CA8518CEBE3DD34837B1E3D5171A6602D6A8BC47009FBDF991ACB5F38FFB3F969AAC77638BDB63BA74CC172DB8329DFA09</hash>
      </hashes>
      <licenses>
        <license>
          <url>http://www.opensource.org/licenses/ms-pl.html</url>
        </license>
      </licenses>
      <purl>pkg:nuget/DotNetOpenAuth.Core@4.0.3.12153</purl>
      <externalReferences>
        <reference type="website">
          <url>http://www.dotnetopenauth.net/</url>
        </reference>
      </externalReferences>
    </component>`

**Reply by stevespringett** [reply]
Only the license url is there not the license ID, name, or expression. Open an issue with CycloneDX.
